syncto/tests/test_functional.py

```diff
@@ -59,6 +59,44 @@ def test_404_endpoint_returns_cors_headers(self):
         self.assertEqual(response.headers['Access-Control-Allow-Origin'],
                          'notmyidea.org')
 
+    def test_client_state_header_with_non_sense_raise_401(self):
+        headers = self.headers.copy()
+        headers['Authorization'] = "BrowserID abcd"
+        headers['X-Client-State'] = "NonSense"
+        patch = mock.patch("syncto.authentication.SyncClient")
+        with patch as sync_client:
+            error = HTTPError()
+            error.response = mock.MagicMock()
+            error.response.status_code = 401
+            error.response.reason = "Unauthorized"
+            error.response.text = ('{"status": "invalid-credentials", '
+                                   '"errors": [{"location": "body", '
+                                   '"name": "", '
+                                   '"description": "Unauthorized"}]}')
+            sync_client.side_effect = error
```

Comment:
I would factorize this into a method to be called by this test case and the next one.

`with self.patched_client(data=data, status=401, reason="Unauthorized")`